query/src/main/scala/filodb/query/LogicalPlan.scala

```diff
@@ -71,23 +71,8 @@ sealed trait PeriodicSeriesPlan extends LogicalPlan {
 
   def replacePeriodicSeriesFilters(filters: Seq[ColumnFilter]): PeriodicSeriesPlan
 
-  /**
-   * Get the data source.
-   * @param earliestRawTime the earliest timestamp of the raw cluster.
-   * @return 0 if all from raw cluster, 1 if all from downsample cluster, 2 if from raw and downsample.
-   */
-  def getDataSource(earliestRawTime: Long): Int = {
-    if (startMs >= earliestRawTime) {
-      0
-    } else if (endMs < earliestRawTime) {
-      1
-    } else {
-      2
-    }
-  }
-
-  def hasAtModifier: Boolean = false
-
+  // The array of all at modifier
+  def atModifierTimestampsWithOffset: Seq[Long] = Seq()
```

Comment:
I see in the `LongTimeRangePlanner` that all `RawSeries` plans (and a couple others) are sent directly to the raw cluster regardless of timestamps. Should that be addressed in this PR?

Reply:
This default implementation might bite us later when a new `PeriodicSeries` is added -- can you do one of either:
- Remove the default implementation.
- Create a `match` function in some utility class that stores this logic for all classes (it's easy to safely give a default implementation for a bunch of classes that way). Something like this:
```
        case lc: LabelCardinality          => materializeLabelCardinalityPlan(lc, qContext)
        ...
        case ts: TsCardinalities           => materializeTSCardinalityPlan(qContext, ts)
        case _: LabelValues |
             _: ApplyLimitFunction |
             _: SeriesKeysByFilters |
             _: ApplyInstantFunctionRaw |
             _: RawSeries |
             _: LabelNames                 => <insert default implementation here>
```

Reply:
can we apply @modifier for RawSeries?

Reply:
Is something like this valid promql?
```
foo{label="value"}[1h] @ 12345
```

FWIW it would be more appropriate to just adjust the query range. Maybe since the logic currently sends all pure RawSeries queries to the raw cluster, we should just reject any with an @modifier in the DS cluster? And we could include a message like "please adjust the time-range"?

Reply:
`foo{label="value"}[1h] ` always goes to the raw cluster.
As discussed, we can fix this later.